### PR TITLE
WIP: fix(app): use BigNum for fungible post condition amount

### LINKED ISSUE #570: [TESTNET BUG] Contract call fails - FungiblePostCondition.amount not correctly decoded
Using `doContractCall` of Connect library with fungible post conditions results in failure in the Blockstack app:

Screenshot_20200804-235156_Firefox_Nightly

This is caused by wrongly decoded amount of the `FungiblePostCondition.amount`

An amount of e.g. `new BigNum(2000)` is handed over as string "07d0" to stacks-transactions-js library while a BigNum is expected (with `toArrayLike` method).

### PR DESCRIPTION
**Pull request by friedger**
This PR
* fixes #570 
* adds a unit test for generating contract call transactions

### CODE REVIEW

**Comment by friedger** [comment]
The newly added unit test fails due to "Can't find module @blockstack/connect". How to setup jest correctly?

File: packages/app/src/common/transaction-utils.ts [hunk 1]
```diff
@@ -30,6 +31,14 @@ export const generateContractCallTx = ({
     return deserializeCV(Buffer.from(arg, 'hex'));
   });
 
+  const conditions = txData.postConditions?.map(c => {
+    const condition = c as unknown;
```
Comment (friedger) [inline packages/app/src/common/transaction-utils.ts:35]: FungiblePostConditions are not exported

**Comment by friedger** [comment]
module problem is solved, now:  TypeError: Cannot read property 'createContext' of undefined

@hstove  any ideas? This prevents app developers to use connect for transaction signing!

**Comment by hstove** [comment]
Closing this - commits in #574

**Pull request closed without merging**